src/components/backgroundselector/BackgroundSelectorDirective.js

```diff
@@ -91,14 +94,26 @@ goog.require('ga_topic_service');
           };
 
           var dereg = scope.$on('gaLayersChange', function(event, newLayers) {
+            getLayersDeffered.resolve();
             updateBgLayer();
             dereg();
```

Comment:
We dereg here, so this function is called only once. Is this wanted? if yes, why is updateGbLayer() still called here?

I'm a little confused about what this directive wants to achieve.